android/src/play/java/com/dooboolab/RNIap/RNIapModule.java

```diff
@@ -399,7 +399,7 @@ public void onPurchaseHistoryResponse(
                   for (int i = 0; i < purchaseHistoryRecordList.size(); i++) {
                     WritableMap item = Arguments.createMap();
                     PurchaseHistoryRecord purchase = purchaseHistoryRecordList.get(i);
-                    item.putString("productId", purchase.getSkus().get(i));
+                    item.putString("productId", purchase.getSkus().get(0));
```

Comment:
Could you kindly give more information on how could this solve the problem?
Wouldn't these changes only return the first `productId`?

Reply:
@hyochan sure.  In the [Google Play Billing Library v 4.0 Release Notes](https://developer.android.com/google/play/billing/release-notes), we learn that Google: 

> Added Purchase#getSkus() and PurchaseHistoryRecord#getSkus(). These replace Purchase#getSku and PurchaseHistoryRecord#getSku which have been removed.

`getSkus()` returns an array of skus, but we don't really have any information _why_ they made this change?  🤷‍♂️

> getSkus
> public ArrayList<String> getSkus ()
> Returns the product Ids.

In my app, I have 4 skus available for purchase, and my test user has purchased each one, but each one is returned as an item in `purchaseHistoryRecordList`, and the `purchase.getSkus()` call return an array with a single item.

The issue with the code using `.get(i)` on the results (`purchase.getSkus().get(i))`) is that once we start looping through the `purchaseHistoryRecordList` and `i` gets incremented, we are trying to `get()` items that _do not exist_.  When this occurs, the method fails and does not throw an error.  This is why the call to `getPurchaseHistory` in version 7.0.x does nothing.  I don't even seen anything in the logcat output when this happens (which is why I can't provide the output, as requested 😉 )

When we use the `.get(0)` on the results of `purchase.getSkus()` then we get results that are _consistent with the previous version of the library_, because the previous version of the library returned a single sku.

I suppose there are other things that we should consider?

1) We could try a nested loop on the results of `getSkus()` inside each item of the `purchaseHistoryRecordList` ?
 ```
for (int i = 0; i < purchaseHistoryRecordList.size(); i++) {
	WritableMap item = Arguments.createMap();
    PurchaseHistoryRecord purchase = purchaseHistoryRecordList.get(i);
    for (int j = 0; j < purchase.getSkus().size(); j++) {
		item.putString("productId", purchase.getSkus().get(j));
		item.putDouble("transactionDate", purchase.getPurchaseTime());
		item.putString("transactionReceipt", purchase.getOriginalJson());
		item.putString("purchaseToken", purchase.getPurchaseToken());
		item.putString("dataAndroid", purchase.getOriginalJson());
		item.putString("signatureAndroid", purchase.getSignature());
		item.putString("developerPayload", purchase.getDeveloperPayload());
		items.pushMap(item);
	}	
}
```
2) Can we wrap this logic in a `try()... catch()` to handle the error better?
3) How can we minimize disruption to developers who are expecting results the same as the earlier versions of the library... and also keep the payload the same as what iOS version returns?  Or do we create a new `getPurchaseHistoryAndroid()` method?

The bottomline, for me, is that when I make the simple change from `.get(i)` to `.get(0)` everything works again, just like it did in 6.3.0... but I'm happy to consider your feedback and help with the best solution for the community.